kotlinx-coroutines-core/native/src/EventLoop.kt

```diff
@@ -14,8 +14,8 @@ internal actual abstract class EventLoopImplPlatform : EventLoop() {
         current.executeAfter(0L, {})// send an empty task to unpark the waiting event loop
     }
 
-    protected actual fun reschedule(now: Long, delayedTask: EventLoopImplBase.DelayedTask) {
-        DefaultExecutor.invokeOnTimeout(now, delayedTask, EmptyCoroutineContext)
+    protected actual fun reschedule(now: ComparableTimeMark, delayedTask: EventLoopImplBase.DelayedTask) {
+        DefaultExecutor.invokeOnTimeout((now - markNow()).inWholeMilliseconds, delayedTask, EmptyCoroutineContext)
```

Comment:
Could you please confirm that a "point of time" value was wrongly passed to a function that expects a "duration of time"? Additionally, the former was in nanos, while the later in milllis.

Reply:
Yes, thanks for noticing!

Reply:
I will fix this in a separate PR